src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/SchemaMetadata.php

```diff
@@ -95,6 +90,24 @@ function(PropertyMetadata $propertyMetadata) {
             $jsonSchema['type'] = $this->type;
         }
 
+        $required = \array_values(
+            \array_map(
+                function(PropertyMetadata $propertyMetadata) {
+                    return $propertyMetadata->getName();
+                },
+                \array_filter(
+                    $this->properties,
+                    function(PropertyMetadata $propertyMetadata) {
+                        return $propertyMetadata->isMandatory();
+                    }
+                )
+            )
+        );
+
+        if (\count($required) > 0 || empty($jsonSchema)) {
```

Comment:
Why the empty check?

Reply:
Mostly explained here https://github.com/sulu/sulu/pull/5666#discussion_r535236464 ... In most cases, `['required' => []]` makes no sense, it's just used to tell the admin ui that the schema is an object instead of an array

Reply:
Would add this as a comment here, otherwise it is not easy to understand.